src/Sentry/Reflection/AssemblyExtensions.cs

```diff
@@ -22,11 +22,28 @@ public static SdkVersion GetNameAndVersion(this Assembly asm)
         {
             var asmName = asm.GetName();
             var name = asmName.Name;
-            var version =
-                asm.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion
-                ?? asmName?.Version?.ToString();
+            string? asmVersion = null;
 
-            return new SdkVersion {Name = name, Version = version};
+            // Note: on full .NET FX, checking the AssemblyInformationalVersionAttribute could throw an exception, therefore
+            // this method uses a try/catch to make sure this method always returns a value
+            try
+            {
+                asmVersion = asm.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion;
+            }
+            catch
+            {
+            }
+
+            // Note: even though the informational version could be "invalid" (e.g. string.Empty), it should
+            // be used for versioning and the software should not fallback to the assembly version string.
+            //
+            // See https://github.com/getsentry/sentry-dotnet/pull/1079#issuecomment-866992216
+            if (asmVersion is null)
```

Comment:
I believe we should fallback to `Version` though. If it's an empty string it could be a bug in the build process that patches the value for example. Having as an empty string is arguably worse than `""`

```suggestion
            if (string.IsNullOrWhiteSpace(asmVersion))
```

So if tests fails with this, I believe changing the tests is the right thing to do